src/shared/bootspec.c

```diff
@@ -52,22 +52,28 @@ void boot_entry_free(BootEntry *entry) {
 }
 
 int boot_entry_load(const char *path, BootEntry *entry) {
+        _cleanup_(boot_entry_free) BootEntry tmp = {};
         _cleanup_fclose_ FILE *f = NULL;
         unsigned line = 1;
-        _cleanup_(boot_entry_free) BootEntry tmp = {};
+        char *b, *c;
         int r;
 
         assert(path);
         assert(entry);
 
-        f = fopen(path, "re");
-        if (!f)
-                return log_error_errno(errno, "Failed to open \"%s\": %m", path);
+        c = endswith_no_case(path, ".conf");
+        if (!c)
+                return log_error("Invalid loader entry filename: %s", path);
```

Comment:
log_error() returns "0". That's not intended here, is it? Should probably be `{ log_error(…); return -EINVAL; }` or so

Reply:
Thank you. Fixed.